agent-control/src/http/client.rs

```diff
@@ -66,17 +66,24 @@ impl HttpClient {
         &self,
         request: Request<Vec<u8>>,
     ) -> Result<HttpResponse<Vec<u8>>, HttpResponseError> {
-        let req = self
+        let req_builder = self
             .client
             .request(request.method().into(), request.uri().to_string().as_str())
             .headers(request.headers().clone())
             .body(request.body().to_vec());
 
-        let res = req
-            .send()
-            .map_err(|err| HttpResponseError::TransportError(err.to_string()))?;
-
-        try_build_response(res)
+        let res = req_builder.send().map_err(from_reqwest_error)?;
+
+        if res.status().is_success() {
+            try_build_response(res)
+        } else {
+            let status_code = res.status();
+            let body = res
+                .bytes()
+                .map_err(|err| HttpResponseError::ReadingResponse(err.to_string()))?
+                .to_vec();
+            Err(HttpResponseError::UnsuccessfulResponse { status_code, body })
```

Comment:
why changing this behaviour ? I see that previously a response with a status code != 2xx was not returning error . I guess clients of this http client are expecting this